core/variables.js

```diff
@@ -275,19 +275,30 @@ Blockly.Variables.generateUniqueName = function(workspace) {
  */
 Blockly.Variables.createVariableButtonHandler = function(
     workspace, opt_callback, opt_type) {
+  var type = opt_type || '';
   // This function needs to be named so it can be called recursively.
   var promptAndCheckWithAlert = function(defaultName) {
     Blockly.Variables.promptName(Blockly.Msg.NEW_VARIABLE_TITLE, defaultName,
       function(text) {
         if (text) {
-          if (workspace.getVariable(text, opt_type)) {
-            Blockly.alert(Blockly.Msg.VARIABLE_ALREADY_EXISTS.replace('%1',
-                text.toLowerCase()),
+            var existing = Blockly.Variables.nameUsedWithAnyType_(text,
+                workspace);
+            if (existing) {
+              var lowerCase = text.toLowerCase();
+              if (existing.type == type) {
+                var msg = Blockly.Msg.VARIABLE_ALREADY_EXISTS.replace('%1',
+                    lowerCase);
```

Comment:
I know you're repeating my prior code, but in retrospect it seems strange to repeat the variable name back to the user's input in the lowercased form.  Ideally, if `existing.name !== text` append something like `' (as "${existing.name}")'` to the variable name. That would make it clear that different name casings map to the same thing.

Reply:
Created #1568